app/uk/gov/hmrc/helptosavefrontend/connectors/CitizenDetailsConnector.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.connectors
+
+import com.google.inject.{ImplementedBy, Singleton}
+import play.api.libs.json.{Json, Reads}
+import uk.gov.hmrc.helptosavefrontend.connectors.CitizenDetailsConnector.CitizenDetailsResponse
+import uk.gov.hmrc.helptosavefrontend.util._
+import uk.gov.hmrc.play.config.ServicesConfig
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext
+
+/**
+  * A connector which connects to the `citizen-details` microservice to obtain
+  * a person's address
+  */
+@ImplementedBy(classOf[CitizenDetailsConnectorImpl])
+trait CitizenDetailsConnector {
+
+  def getDetails(nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext): Result[CitizenDetailsResponse]
+}
+
+object CitizenDetailsConnector {
+
+  case class Address(line1: String, line2: String, line3: String, line4: String,
+                     line5: String, postcode: String, country: String)
+
+  case class CitizenDetailsResponse(address: Address)
```

Comment:
I put it here because it was specific to the `CitizensDetailsConnector` rather than being used by the rest of the project. This `Address` is specific to the format returned by citizens details - at the moment in `UserInfo` the address is modelled by `List[String]`. I didn't want confusion as to why in some places we were using one data structure and in others another.